utilities/transactions/write_prepared_txn_db.cc

```diff
@@ -319,11 +331,30 @@ std::vector<Status> WritePreparedTxnDB::MultiGet(
     const std::vector<Slice>& keys, std::vector<std::string>* values) {
   assert(values);
   size_t num_keys = keys.size();
+  std::vector<Status> stat_list(num_keys);
+
+  if (options.io_activity != Env::IOActivity::kUnknown &&
+      options.io_activity != Env::IOActivity::kMultiGet) {
+    Status s = Status::InvalidArgument(
+        "Can only call MultiGet with `ReadOptions::io_activity` is "
+        "`Env::IOActivity::kUnknown` or `Env::IOActivity::kMultiGet`");
+
+    for (size_t i = 0; i < num_keys; ++i) {
+      stat_list[i] = s;
+    }
+    return stat_list;
+  }
+
+  ReadOptions complete_read_options(options);
+  if (complete_read_options.io_activity == Env::IOActivity::kUnknown) {
+    complete_read_options.io_activity = Env::IOActivity::kMultiGet;
+  }
+
   values->resize(num_keys);
 
-  std::vector<Status> stat_list(num_keys);
   for (size_t i = 0; i < num_keys; ++i) {
-    stat_list[i] = this->Get(options, column_family[i], keys[i], &(*values)[i]);
+    stat_list[i] = this->GetImpl(complete_read_options, column_family[i],
```

Comment:
Refactory 4:
Previous:
```
MultiGet()
-> virtual DB::Get(string*)
-> virtual DB::Get(PinnableSlice*) [overridden by WriteUnpreparedTxnDB]
-> WriteUnpreparedTxnDB::Get(PinnableSlice*) 
```

Now:
```
MultiGet()
-> WriteUnpreparedTxnDB::GetImpl(string*) [WriteUnpreparedTxnDB::GetImpl(string*) basically duplicates virtual DB::Get(string*) since I don't want to make a virtual DB::GetImpl() triggering more refactoring for name collision on XXDBImpl::GetImpl(). Some redundancy but keeping the scope of change small for now.]
-> WriteUnpreparedTxnDB::GetImpl(PinnableSlice*)
```